lib/features/public_asset/domain/usecase/clean_up_public_assets_interactor.dart

```diff
@@ -0,0 +1,117 @@
+        identityId,
+        identityActionType,
+        publicAssetsInIdentityArguments);
+
+      await Rx.merge<Either<Failure, Success>>([
+        if (publicAssetIds.publicAssetsIdsToBeDereferenced.isNotEmpty)
+          removeIdentityFromPublicAssetsInteractor.execute(
+            session,
+            accountId,
+            identityId: identityId,
+            publicAssetIds: publicAssetIds.publicAssetsIdsToBeDereferenced),
+        if (publicAssetIds.publicAssetsIdsToBeDestroyed.isNotEmpty)
+          deletePublicAssetsInteractor.execute(
+            session,
+            accountId,
+            publicAssetIds: publicAssetIds.publicAssetsIdsToBeDestroyed),
+        if (publicAssetIds.publicAssetsIdsToBeReferenced.isNotEmpty)
+          addIdentityToPublicAssetsInteractor.execute(
+            session,
+            accountId,
+            identityId: identityId,
+            publicAssetIds: publicAssetIds.publicAssetsIdsToBeReferenced),
+      ]).last;
+
+      yield Right(CleanUpPublicAssetsSuccessState());
+
+    } catch (exception) {
+      logError('CleanUpPublicAssetsInteractor::execute():error: $exception');
+      yield Left(CleanUpPublicAssetsFailureState(exception: exception));
+    }
+  }
+
+  ({
+    List<PublicAssetId> publicAssetsIdsToBeDestroyed,
+    List<PublicAssetId> publicAssetsIdsToBeReferenced,
+    List<PublicAssetId> publicAssetsIdsToBeDereferenced, 
+  }) _filterPublicAssetsIdsForCleanUp(
+    IdentityId identityId,
+    IdentityActionType identityActionType,
+    PublicAssetsInIdentityArguments publicAssetsInIdentityArguments,
+  ) {
+    final htmlSignature = publicAssetsInIdentityArguments.htmlSignature;
+    final publicAssetIdsInSignature = htmlSignature.publicAssetIdsFromHtmlContent;
+    log('CleanUpPublicAssetsInteractor::publicAssetIdsInSignature: $publicAssetIdsInSignature');
+    final preExistingPublicAssetIds = publicAssetsInIdentityArguments.preExistingPublicAssetIds;
+    log('CleanUpPublicAssetsInteractor::preExistingPublicAssetIds: $preExistingPublicAssetIds');
+    final newlyPickedPublicAssetIds = publicAssetsInIdentityArguments.newlyPickedPublicAssetIds;
+    log('CleanUpPublicAssetsInteractor::newlyPickedPublicAssetIds: $newlyPickedPublicAssetIds');
+    final deletedpreExistingPublicAssetIds = preExistingPublicAssetIds
+      .where(publicAssetIdsInSignature.notContains)
+      .toList();
+    log('CleanUpPublicAssetsInteractor::deletedpreExistingPublicAssetIds: $deletedpreExistingPublicAssetIds');
+    final notIncludednewlyPickedPublicAssetIds = newlyPickedPublicAssetIds
+      .where(publicAssetIdsInSignature.notContains)
+      .toList();
+    log('CleanUpPublicAssetsInteractor::notIncludednewlyPickedPublicAssetIds: $notIncludednewlyPickedPublicAssetIds');
+    final pastedPublicAssetIds = publicAssetIdsInSignature
+      .where(preExistingPublicAssetIds.notContains)
+      .where(newlyPickedPublicAssetIds.notContains)
+      .toList();
```

Comment:
Why not use a `where` with `&` 2 conditions?

Reply:
it's cleaner and more readable this way

Reply:
As you wrote, it will run 2 loops to check. We should optimize it.

Reply:
Done